### PR TITLE
Add all featured image sizes to events

### LINKED ISSUE #954: Change event thumbnail image size to medium
The web app scales the event thumbnails to the page width. 150px is not enough. We can use the medium (or even large) thumbnails instead.

https://codex.wordpress.org/Function_Reference/the_post_thumbnail_url#Thumbnail_Sizes

### PR DESCRIPTION
**Pull request by svenseeberg**
* For events, the web app uses the featured image with larger
  resolution than the 150px * 150px thumbnail. Therefore we
  need larger images. This commit adds the other preview sizes
  (medium, large) and also the full size image URL to the
  events APIv3 endpoint.
* Fixes #954 
* Adds the following object to each event in the APIv3, if an
  featured image exists. Otherwise, the `featured_image` is `null`.
```
    "featured_image": {
        "medium": String, // URL to 300 px image
        "large": String, // URL to 768px image
        "full": String, // URL to full sized image
    }
```
### Your checklist for this pull request
- [x] Check if patches need to be applied.
  - [ ] Patch 0001 - WPML
  - [ ] Patch 0002 - Broken Link Checker
  - [ ] Patch 0003 - CMS Tree View
  - [ ] Patch 0004 - Sitepress
- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *develop*.
- [x] Check the commit's or even all commits' message styles matches your requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### CODE REVIEW

**Comment by Schedulaar** [comment]
We might consider adding that to both pages and disclaimer as well, but that's not required as of now.

**Comment by maxammann** [comment]
Are we sure those resolutions are enough?

**Comment by svenseeberg** [comment]
@maxammann meaning you want more than 4 different resolutions? I'd say that 4 is enough.

**Comment by maxammann** [comment]
I think we should use this change to improve the API even further. Or at least make it not breaking in the future. I lookup up how 2 other APIs are doing it:

- https://www.contentful.com/developers/docs/references/images-api/#/reference
- https://api.imgur.com/models/image/

I think it would make sense to deliver further information like: `mimeType`, `width`, `height`, `fileSize`, `altTag`/`description`(, `hash`)

Especially `altTag` and the actual size are important.

```
"featured_image": {
        "300": {
            "url": "",
            "width": 300,
            "height": 300,
            "mimeType": "image/jpg",
            "size": 156478,
            "description": "blub"
        }
    }
```
or:
```
"featured_image": [
        {
            "url": "",
...
        }
 ]
```

At least that structure of using objects would make sense to extend it more in the future.

**Comment by svenseeberg** [comment]
The mime type and alt text are the same for each resolution. And is the resolution in pixels really useful? Therefore:
```
    "featured_image": {
        "mimetype": String // "image/png", "image/jpg", etc
        "description": String, // User entered string describing the image
        "medium": String, // URL to 300 px image
        "large": String, // URL to 768px image
        "full": String, // URL to full sized image
    }
```

**Comment by maxammann** [comment]
Yes the **width** and **height** can be used to avoid page reflows. This means we can reserve space on the website in order to avoid pushing text around.

**size** can be used to let javascript decide which resolution the browser should load so it also makes sense.

I think **mimetype** should be next to witdth and height because it could make sense to deliver images of different sizes in different file formats like png, webp etc.
For future use it could also make sense to deliver images in different formats. Modern browsers can use webp and older jpegs. Therefore we actually need a different structure to support this:

```
"featured_image": {
        "description": String, // User entered string describing the image
        "files": [
            {
                "mimetype": String // "image/png", "image/jpg", etc
                "url": "",
                "width: 10,
                "height: 10,
                "size": 100, size in kilobytes
            }
        ]
}
```
I think an array is more flexible because defining a name for a format is quite difficult. OR:
```
"featured_image": {
        "description": String, // User entered string describing the image
        "320": [
            {
                "mimetype": String // "image/png", "image/jpg", etc
                "url": "",
                "width: 320,
                "height: 10,
                "size": 100, size in kilobytes
            }
        ],
        "480": [
            {
                "mimetype": String // "image/png", "image/jpg", etc
                "url": "",
                "width: 480,
                "height: 10,
                "size": 100, size in kilobytes
            }
        ],
        "800": [
            {
                "mimetype": String // "image/png", "image/jpg", etc
                "url": "",
                "width: 800,
                "height: 10,
                "size": 100, size in kilobytes
            }
        ]
}
```
The 320, 480 and 800 are the sizes of the **viewport**.

To summarize we need the following from the api for images:
- A description of the image
- A size of each candidate of images
- A bounding box for each candidate
- The mime type of each candidate in order to "ask" browsers which format they support.

[Adaptive Imaging](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images)

**Comment by svenseeberg** [comment]
Wordpress does not store the size of images. We can read them from hard disk, when creating the API request answer. That would be very slow. Are the sizes really NEEDED? @maxammann

**Comment by svenseeberg** [comment]
There is one other option: We can set our own resolution with forced cropping. Then we could get rid of the width and height attributes altogether, because we have defined values.

**Comment by svenseeberg** [comment]
But accident I found a function that returns the image resolution. No luck with the size so far. Current status:
```
"featured_image": {
  "description":null,
  "mimetype":"image\/png",
  "thumbnail":[{
    "url":"http:\/\/integreat\/augsburg\/wp-content\/uploads\/sites\/2\/2019\/04\/image-150x150.png",
    "height":150,
    "width":150
  }],
  "medium":[{
    "url":"http:\/\/integreat\/augsburg\/wp-content\/uploads\/sites\/2\/2019\/04\/image-300x164.png",
    "height":164,
    "width":300
  }],
  "large":[{
    "url":"http:\/\/integreat\/augsburg\/wp-content\/uploads\/sites\/2\/2019\/04\/image-1024x559.png",
    "height":360,
    "width":660
  }]
}
```

**Comment by maxammann** [comment]
I think we can live without sizes. I mean we can still estimate the size and choose the corrosponding one.
There is no problem with the widths, right?

**Comment by svenseeberg** [comment]
image
